plonk/src/proof_system/prover.rs

```diff
@@ -919,6 +921,29 @@ impl<E: PairingEngine> Prover<E> {
                 )
             })
             .collect();
+
+        // mask splitting polynomials
+        // t'(X) = t(X) - b_last + b_now * X^(n+2)
+        // with t'_lowest(X) = t_lowest(X) - 0 + b_now * X^(n+2)
+        // and t'_highest(X) = t_highest(X) - b_last
+        let mut last_randomizer = E::Fr::zero();
+        split_quot_polys
+            .iter_mut()
+            .take(num_wire_types - 1)
+            .for_each(|poly| {
+                let now_randomizer = E::Fr::rand(prng);
+
+                let mut coeffs = vec![E::Fr::zero(); n + 3];
+                coeffs[0] = last_randomizer.neg();
+                coeffs[n + 2] = now_randomizer;
+                *poly += &DensePolynomial::from_coefficients_vec(coeffs);
```

Comment:
Is it possible to directly change `poly`'s 2 coefficients rather than adding an O(n)-sized polynomial? Same as the last splitting poly.

Reply:
very good idea. Mine is easier to read but very wasteful in memory.